langchain4j-agui/src/main/java/dev/langchain4j/agent/agui/ToolMapper.java

```diff
@@ -0,0 +1,115 @@
+package dev.langchain4j.agent.agui;
+
+import com.agui.core.tool.Tool;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.model.chat.request.json.JsonArraySchema;
+import dev.langchain4j.model.chat.request.json.JsonIntegerSchema;
+import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
+import dev.langchain4j.model.chat.request.json.JsonStringSchema;
+
+/**
+ * Utility class for converting agui tool definitions to LangChain4j tool specifications.
+ * <p>
+ * ToolMapper provides conversion functionality between agui's Tool format and LangChain4j's
+ * ToolSpecification format, enabling seamless integration of function calling capabilities
+ * between the two frameworks. The mapper handles tool metadata, parameter definitions,
+ * and JSON schema conversion for proper tool registration with LangChain4j models.
+ * <p>
+ * The conversion process includes:
+ * <ul>
+ * <li>Tool name and description mapping</li>
+ * <li>Parameter schema conversion to JSON Object Schema</li>
+ * <li>Type validation and mapping for tool parameters</li>
+ * <li>Property description preservation</li>
+ * </ul>
+ * <p>
+ * Current limitations:
+ * <ul>
+ * <li>Only string parameter types are currently supported</li>
+ * <li>Complex nested parameter structures are not yet implemented</li>
+ * </ul>
+ * <p>
+ * This class is stateless and thread-safe, making it suitable for use in
+ * concurrent environments and as a singleton or utility class.
+ *
+ * @author Pascal Wilbrink
+ */
+public class ToolMapper {
+
+    /**
+     * Converts an agui Tool to a LangChain4j ToolSpecification.
+     * <p>
+     * This method performs a comprehensive conversion of tool definitions, mapping
+     * all essential tool metadata including name, description, and parameter specifications.
+     * The parameter conversion uses JSON Object Schema format as required by LangChain4j
+     * for proper function calling integration with language models.
+     * <p>
+     * The conversion process:
+     * <ul>
+     * <li>Directly maps tool name and description</li>
+     * <li>Converts tool parameters to JsonObjectSchema via {@link #toJsonObjectSchema}</li>
+     * <li>Builds a complete ToolSpecification ready for LangChain4j consumption</li>
+     * </ul>
+     * <p>
+     * The resulting ToolSpecification can be used with LangChain4j's function calling
+     * mechanisms, allowing language models to understand and invoke the tool with
+     * properly typed parameters.
+     *
+     * @param tool the agui Tool to convert
+     * @return a LangChain4j ToolSpecification with name, description, and parameter schema
+     * @throws RuntimeException if the tool contains unsupported parameter types
+     */
+    public ToolSpecification toLangchainTool(final Tool tool) {
```

Comment:
```suggestion
    public ToolSpecification toLangChain4jTool(final Tool tool) {
```
Please update all instances of Langchain -> LangChain4j